### ISSUE TITLE
resolving issue connecting to localhost on macOS

### ISSUE THREAD

**Issue by vrischmann** [gocql/gocql#997]
### What version of Cassandra are you using?

3.11.1

### What version of Gocql are you using?

Commit 900c1e0053b3e2c7f6d7acd8f0f6bd7a599a3735

### What did you do?

Trying to connect to `localhost`.

### What did you expect to see?

A valid session

### What did you see instead?

gocql (or Go rather) resolves `localhost` to an IPv6 address and the connection fails.

    $ go run -tags gocql_debug -tags netdns gocql.go -port 9042                                                                                                                                            
    2017/10/25 15:45:15 <nil> gocql: unable to create session: unable to discover protocol version: dial tcp [::1]:9042: getsockopt: connection refused

This is the code used to test:

```go
package main

import (
	"flag"
	"log"

	"github.com/gocql/gocql"
)

func main() {
	flPort := flag.Int("port", 0, "")

	flag.Parse()

	cluster := gocql.NewCluster("localhost")
	cluster.Port = *flPort

	session, err := cluster.CreateSession()
	log.Println(session, err)
}
```

I'm testing against a single Cassandra 3.11.1 node downloaded [from here](http://www.apache.org/dyn/closer.lua/cassandra/3.11.1/apache-cassandra-3.11.1-bin.tar.gz) and simply running it without touching anything.

If I replace `localhost` with `127.0.0.1` it works fine.

I'm pretty sure this worked some time ago but it's been a long time since I used a local Cassandra so I can't say for sure since when it fails.

I'm also not sure if something changed with macOS, maybe they do something weird now.

**Comment by Zariel** [comment]
I've ran into this locally for a while and have a work around for running tests locally https://github.com/gocql/gocql/blob/843f6b1d2288839f5d00f92be11cbbebd600ba51/control.go#L94

**Comment by vrischmann** [comment]
Yep that works, thanks. Would you accept a PR that reads that setting from a environment variable, something like `GOCQL_HOST_LOOKUP_PREFER_V4` ? It would be more convenient than having to modify the code directly, at least for me

**Comment by Zariel** [comment]
sure

**Comment by Zariel** [comment]
fixed via #1000
